src/main/topLevelPages/myDashboard/myMemberships/MyMembershipsDialog.tsx

```diff
@@ -66,57 +58,14 @@ const MyMembershipsDialog = ({ open, onClose }: MyJourneysDialogProps) => {
 
   const myTopLevelMemberships = useMemo(() => data?.me.spaceMemberships.filter(space => space?.level === 0), [data]);
 
-  // As the query returns all levels of memberships, we're using a map for ease of access to the data of each membership
-  // without having to iterate over the array each time or making redundant queries for the details of every subspace
-  const allMyMembershipsMap: Record<string, MembershipProps> = useMemo(
-    () =>
-      (data?.me.spaceMemberships ?? []).reduce((acc, item) => {
-        acc[item.id] = item;
-        return acc;
-      }, {}),
-    [data]
-  );
-
-  const getMembership = (id: string) => allMyMembershipsMap[id];
-
   return (
-    <DialogWithGrid open={open} onClose={onClose} columns={12}>
+    <DialogWithGrid open={open} onClose={onClose} columns={8}>
       <DialogHeader icon={<SpaceIcon />} title={t('pages.home.sections.myMemberships.title')} onClose={onClose} />
       <DialogContent>
         {loading && <Loading />}
-        <Gutters disablePadding>
+        <Gutters disablePadding disableGap>
           {myTopLevelMemberships?.map(space => (
-            <PageContentBlock
-              row
-              flexWrap="wrap"
-              sx={{
-                background: theme => theme.palette.background.default,
-                alignItems: 'start',
-              }}
-            >
-              <SpaceCard
-                banner={space.profile.cardBanner}
-                displayName={space.profile.displayName}
-                vision={space.context?.vision ?? ''}
-                tagline={space.profile.tagline}
-                journeyUri={space.profile.url}
-                tags={space.profile.tagset?.tags ?? []}
-                membersCount={getMetricCount(space.metrics, MetricType.Member)}
-                spaceVisibility={space.account.license.visibility}
-              />
-              <GridItem columns={9}>
-                <Gutters row disablePadding flexGrow={1} flexWrap="wrap">
-                  <GridProvider columns={8}>
-                    {space.subspaces?.filter(isJourneyMember).map(subspace => (
```

Comment:
⚠️ This check is important as if you're part of a Space, it doesn't mean you are part of the subspaces. In this dialog we should show only the spaces/subspaces/subsbuspaces the user is actually a member of.

In case of a private subspace (that the user is not a member of) currently the entire query is failing.

Reply:
fixed with 81d71b5152ec37215b7616d6ae26e9bf9921eb88